src/libIR2Vec.cpp

```diff
@@ -9,30 +9,31 @@
 #include "IR2Vec.h"
 #include "Symbolic.h"
 #include "utils.h"
-
 #include "llvm/IR/Module.h"
+#include "llvm/Support/CommandLine.h"
 
-int IR2Vec::Embeddings::generateEncodings(llvm::Module &M,
-                                          IR2Vec::IR2VecMode mode,
-                                          std::string vocab, char level,
-                                          std::ostream *o, int cls, float WO,
-                                          float WA, float WT) {
+int IR2Vec::Embeddings::generateEncodings(
+    llvm::Module &M, IR2Vec::IR2VecMode mode, std::string vocab, char level,
+    std::string fname, std::ostream *o, int cls, float WO, float WA, float WT) {
   IR2Vec::vocab = vocab;
   IR2Vec::level = level;
   IR2Vec::cls = cls;
   IR2Vec::WO = WO;
   IR2Vec::WA = WA;
   IR2Vec::WT = WT;
-
+  IR2Vec::fname = fname;
   if (mode == IR2Vec::IR2VecMode::FlowAware) {
     IR2Vec_FA FA(M);
     FA.generateFlowAwareEncodings(o);
     instVecMap = FA.getInstVecMap();
     funcVecMap = FA.getFuncVecMap();
     pgmVector = FA.getProgramVector();
-  }
-
-  else if (mode == IR2Vec::IR2VecMode::Symbolic) {
+  } else if (mode == IR2Vec::IR2VecMode::Symbolic && !fname.empty()) {
```

Comment:
suppose user doesnt gave any function name and want symbolic embeddings, will this work?